sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStoreCache.java

```diff
@@ -216,7 +222,13 @@ public void clearTask(String taskName) throws StateStoreException {
     public Optional<TaskInfo> fetchTask(String taskName) throws StateStoreException {
         RLOCK.lock();
         try {
-            return Optional.ofNullable(nameToTask.get(taskName));
+            if (nameToTask.containsKey(taskName)) {
```

Comment:
Could just do `nameToTask.get()` and check for a null value. Single map lookup instead of two.